pkg/controller/ecr/repositorypolicy/controller.go

```diff
@@ -0,0 +1,214 @@
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: awsecr.New(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client ecr.RepositoryPolicyClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	response, err := e.client.GetRepositoryPolicyRequest(&awsecr.GetRepositoryPolicyInput{
+		RegistryId:     cr.Spec.ForProvider.RegistryID,
+		RepositoryName: cr.Spec.ForProvider.RepositoryName,
+	}).Send(ctx)
+
+	if err != nil {
+		if ecr.IsRepoNotFoundErr(err) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(ecr.IsPolicyNotFoundErr, err), errGet)
+	}
+
+	policyData, err := e.formatRepositoryPolicy(cr)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	if response.PolicyText != nil {
+		cr.Status.AtProvider.PolicyText = *response.PolicyText
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	ecr.LateInitializeRepositoryPolicy(&cr.Spec.ForProvider, response)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, awsclient.Wrap(err, errSpecUpdate)
+		}
+	}
+
+	cr.SetConditions(xpv1.Available())
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: ecr.IsRepositoryPolicyUpToDate(policyData, &cr.Status.AtProvider.PolicyText),
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(xpv1.Creating())
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errStatusUpdate)
+	}
+	policyData, err := e.formatRepositoryPolicy(cr)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+	_, err = e.client.SetRepositoryPolicyRequest(ecr.GenerateSetRepositoryPolicyInput(&cr.Spec.ForProvider, policyData)).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, awsclient.Wrap(err, errCreate)
+	}
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
```

Comment:
```suggestion
	return managed.ExternalCreation{}, awsclient.Wrap(err, errCreate)
```

Is there a specific reason that requires us to issue a spec update?

Reply:
Thought there was but looking again can't seem to figure out why will remove that.